lib/fluent/log.rb

```diff
@@ -163,21 +163,22 @@ def logdev=(logdev)
     def format=(fmt)
       return if @format == fmt
 
+      @time_format = '%Y-%m-%d %H:%M:%S %z'
+      @time_formatter = Strftime.new(@time_format) rescue nil
+
       case fmt
       when :text
         @format = :text
-        @time_format = '%Y-%m-%d %H:%M:%S %z'
         @formatter = Proc.new { |type, time, level, msg|
           r = caller_line(type, time, @depth_offset, level)
           r << msg
           r
         }
       when :json
         @format = :json
-        @time_format = '%Y-%m-%d %H:%M:%S %z'
         @formatter = Proc.new { |type, time, level, msg|
           r = {
-            'time' => time.strftime(@time_format),
+            'time' => @time_formatter ? @time_formatter.exec(time) : time.strftime(@time_format),
```

Comment:
How about defining method for `@time_formatter ? @time_formatter.exec(time) : time.strftime(@time_format)`?

Reply:
Done.